tests/checkout/tree.c

```diff
@@ -1229,7 +1234,9 @@ void test_checkout_tree__case_changing_rename(void)
 
 	cl_git_pass(git_index_write(index));
 
+	cl_git_pass(git_libgit2_opts(GIT_OPT_ENABLE_STRICT_OBJECT_CREATION, 0));
 	cl_git_pass(git_index_write_tree(&tree_id, index));
+	cl_git_pass(git_libgit2_opts(GIT_OPT_ENABLE_STRICT_OBJECT_CREATION, 1));
```

Comment:
If we never verify anything that comes from an index, that means you can avoid the strict object creation check very easily by chance by using an unowned index.

I would say the "sentiment" is about not erroring out when reading in data that's already in the repository rather than allowing untouched entries to point to missing objects.

Reply:
This is icky, though.  Since you're about to go on holiday, I took the liberty of dropping this commit and fixing the index so that it's not pointing to a nonexistent blob.